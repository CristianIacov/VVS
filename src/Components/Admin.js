
import React, { Component } from 'react';
class  Admin extends React.Component {
     constructor(props){
  super(props);
  this.state={

  }}
OnPress = () => {
	 fetch('http://localhost:3001/shutdown',{
    method: 'get',
    headers: {'Content-Type':'application/json'}
  })
  .then(response=>response.json())
  .then(data => {
       
        window.alert('Shutting down');

        this.props.onRouteChange('stoppedserver');
        this.props.stopServer(true);
  })

}
OnPress2 = () => {
	 fetch('http://localhost:3001/maintenance',{
    method: 'get',
    headers: {'Content-Type':'application/json'}
  })
  .then(response=>response.json())
  .then(data => {
       
        window.alert('Maintenance');

        this.props.onRouteChange('maintenance');
        this.props.stopServer(true);
  })

}
render()
{
	  return (

<div>
 

<p> Admin page </p>
<p> Server state is RUNNING </p>
<p> Server is running on port 3001 </p>
<input onClick={this.OnPress} type="button" id="stop" value="Stop server"  />
<input onClick={this.OnPress2} type="button" id="maintenance" value="Maintenance"  />
</div>

	  	);

	}
}
	export default Admin;