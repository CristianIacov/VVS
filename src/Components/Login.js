import React, { Component } from 'react';
import './Login.css';
class  Login extends React.Component {
     constructor(props){
  super(props);
  this.state={
    signInName:'',
    signInPassword:''
  }}
  onNameChange = (event) => {
  this.setState({signInName: event.target.value})
}
onPasswordChange = (event) => {
  this.setState({signInPassword: event.target.value})
}
onSubmitLogin1= () => {
  fetch('http://localhost:3001/signin',{
    method: 'post',
    headers: {'Content-Type':'application/json'},
    body: JSON.stringify({
      name:this.state.signInName,
      password: this.state.signInPassword
    })
  })
  .then(response=>response.json())
  .then(data => {
       
          if(data  === 'success'){
          

          this.props.onRouteChange('visualize');
          window.alert('You are logged in');
        }
        else
        {
        	if(data === 'admin')
        	{
        		this.props.setAdmin(true);
        		 this.props.onRouteChange('admin');
        		window.alert('You are logged in as a Admin');
        	}
        	else

        	window.alert('Wrong credentials');
        }
       
        
        	
  })
 
}

changeRoute= () => 
	 this.props.onRouteChange('logged');
  	render(){
		
  return (
<form>
<ul className="form-style-1">
<h1> Login </h1>

<li>
    <label>Username <span className="required">*</span></label>
        <input onChange={this.onNameChange} type="name"  className="field-long"   />
        <label>Password <span className="required">*</span></label>
        <input  onChange={this.onPasswordChange} type="password"  className="field-long"     />

</li>
   <li>
        
        
        
    </li>
<li>

  <input onClick={this.onSubmitLogin1} type="button" value="Submit"  />
  
 </li>  

</ul>
</form>	
    
  );
}}

export default Login;