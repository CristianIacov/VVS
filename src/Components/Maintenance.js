
import React, { Component } from 'react';
class  Maintenance extends React.Component {
     constructor(props){
  super(props);
  this.state={

  }}
    onButtonPress2 = () =>{
  	    
  	    this.props.onRouteChange('home');
        this.props.stopServer(false);

  }
render()
{
  	  return (

<div>
 <p>Please note that we will be performing important server maintenance in a few minutes, during which time the server will be unavailable for approximately 5-10 minutes.</p>
 <p>If you are in the middle of something important, please save your work or hold off on any critical actions until we are finished. </p>
 <input onClick={this.onButtonPress2} type="button" value="Stop maintenance"  />
</div>

	  	);

	}
}
	export default Maintenance;