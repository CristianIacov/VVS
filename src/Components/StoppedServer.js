
import React, { Component } from 'react';
class  StoppedServer extends React.Component {
     constructor(props){
  super(props);
  this.state={

  }}

  onButtonPress = () =>{
  	    
  	    this.props.onRouteChange('home');
        this.props.stopServer(false);

  }
render()
{
  	  return (

<div>
 <p> Serverul este oprit </p>
 <input onClick={this.onButtonPress} type="button" value="Start server"  />
</div>
 
	  	);

	}
}
	export default StoppedServer;