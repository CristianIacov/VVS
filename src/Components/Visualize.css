
.column {
  float: left;
  width: 33.33%;
  padding: 60px;

   font-family: "avenir next";
    text-align: right;
    
}

/* Clear floats after image containers */
.row::after {
	 font-family: "avenir next";
	 text-align: right;
  content: "";
  clear: both;
  display: table;
}

