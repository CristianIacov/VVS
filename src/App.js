
import React, { Component } from 'react';
import {render} from 'react-dom';
import Login from './Components/Login';
import Visualize from './Components/Visualize';
import Admin from './Components/Admin';
import StoppedServer from './Components/StoppedServer';
import Maintenance from './Components/Maintenance';
class App extends React.Component{

  constructor(){
    super();
    this.state={
      route: 'home',
      isAdmin: false,
      isShutdown: false,
        user: {
        

  name: ''
 
    }
  }
}
stopServer = (data) =>
  this.setState({isShutdown : data
  });
setAdmin = (data) =>
  this.setState({isAdmin : data
  });
 
   loadUser = (data) =>{
   this.setState({user:
   {  name: data.name
   }})}

    onRouteChange = (route) => {
    // if( route === 'signout'){
    //   this.setState({isSignedIn: false})

    // }
    // else 
      if (route === 'home')
    {
      this.setState({isHome: true})
    }
   
    this.setState({route: route});
  }
  render(){
  const {route} = this.state;
  const {isShutdown} = this.state;
 return (
  
  <div>
  {
    
      route ==='home'?
      <Login setAdmin={this.setAdmin} onRouteChange={this.onRouteChange}/>
      :
      (route === 'visualize' && isShutdown === false?

      <Visualize onRouteChange={this.onRouteChange}/>
      
      :
     (route === 'admin'  && isShutdown === false?

    <Admin onRouteChange={this.onRouteChange} stopServer = {this.stopServer}/>
      :
      (route === 'stoppedserver'?

        <StoppedServer stopServer={this.stopServer} onRouteChange={this.onRouteChange}/>
        :
         (route === 'maintenance'?
          <Maintenance stopServer={this.stopServer} onRouteChange={this.onRouteChange}/>
          :
      <p> Serverul este oprit </p>
      ))))

    
    }
  </div>
    );

}
}

export default App