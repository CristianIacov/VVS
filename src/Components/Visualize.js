import React, { Component } from 'react';
import amgheghe from '../Imagini/amgheghe.jpg';
import m8 from '../Imagini/m8.jpg';
import rsq8 from '../Imagini/rsq8.jpg';
import './Visualize.css';
class  Visualize extends React.Component {
	     constructor(props){
  super(props);
}
SubmitAmg = () => {
  fetch('http://localhost:3001/amg',{
    method: 'get',
    headers: {'Content-Type':'application/json'}
  })
  .then(response=>response.json())
  .then(data => {
       
    this.props.onRouteChange(data);
        
        	
  })
 
}

SubmitRsq8 = () => {
  fetch('http://localhost:3001/rsq8',{
    method: 'get',
    headers: {'Content-Type':'application/json'}
  })
  .then(response=>response.json())
  .then(data => {
       
    this.props.onRouteChange(data);
        
        	
  })
 
}
render()
{
  return (

  <div className="row">
  <div className="column"> 
<a onClick={this.SubmitAmg}className=" hover-bg-light-green">
<img  src={m8} /	>
</a>

</div>
<div className="column">
<a onClick={this.SubmitRsq8} className=" hover-bg-light-green">
<img  src={rsq8}/	>
</a>

</div>

</div>


  );
}
}

export default Visualize