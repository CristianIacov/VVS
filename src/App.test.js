import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Login from './Components/Login.js'
import { shallow } from 'enzyme';
import Admin from './Components/Admin.js'
import App from './App.js'
import Visualize from './Components/Visualize.js'
import Maintenance from './Components/Maintenance.js'
import StoppedServer from './Components/StoppedServer.js'
import setAdmin from './App.js'

jest.mock("./App");
test('asda',() => {
	expect(2+2).toBe(4);
}
);
test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(<Admin />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(<Maintenance />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders correctly', () => {
  const tree = renderer.create(<StoppedServer />).toJSON();
  expect(tree).toMatchSnapshot();
});