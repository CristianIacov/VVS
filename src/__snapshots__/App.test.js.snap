// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <ul
    className="form-style-1"
  >
    <h1>
       Login 
    </h1>
    <li>
      <label>
        Username 
        <span
          className="required"
        >
          *
        </span>
      </label>
      <input
        className="field-long"
        onChange={[Function]}
        type="name"
      />
      <label>
        Password 
        <span
          className="required"
        >
          *
        </span>
      </label>
      <input
        className="field-long"
        onChange={[Function]}
        type="password"
      />
    </li>
    <li />
    <li>
      <input
        onClick={[Function]}
        type="button"
        value="Submit"
      />
    </li>
  </ul>
</form>
`;

exports[`renders correctly 2`] = `
<div>
  <p>
     Admin page 
  </p>
  <p>
     Server state is RUNNING 
  </p>
  <p>
     Server is running on port 3001 
  </p>
</div>
`;

exports[`renders correctly 3`] = `null`;

exports[`renders correctly 4`] = `
<div>
  <p>
    Please note that we will be performing important server maintenance in a few minutes, during which time the server will be unavailable for approximately 5-10 minutes.
  </p>
  <p>
    If you are in the middle of something important, please save your work or hold off on any critical actions until we are finished. 
  </p>
  <input
    onClick={[Function]}
    type="button"
    value="Stop maintenance"
  />
</div>
`;

exports[`renders correctly 5`] = `
<div>
  <p>
     Serverul este oprit 
  </p>
  <input
    onClick={[Function]}
    type="button"
    value="Start server"
  />
</div>
`;
